app/Services/BaseService.php

```diff
@@ -24,7 +26,45 @@ abstract class BaseService
     /**
      * The contact object.
      */
-    public Contact $contact;
+    public ?Contact $contact = null;
+
+    /**
+     * The contact object.
+     */
+    public ?Group $group = null;
+
+    /**
+     * Dependencies between permissions.
+     *
+     * @var array<string,array<string>>
+     */
+    private static array $dependencies = [
```

Comment:
Would `permissionDependencies` be a better name for this? `dependencies` alone seems a bit off.

Reply:
done, thanks for the suggestion